### PR TITLE
Add a "substitutions" parameter to operations

### PR DESCRIPTION
**Pull request by achew22**
The new "substitutions" parameter is passed to a call to
actions.expand_template on the template (json or yaml) that is passed
into k8s_object.

Fixes: #244

### CODE REVIEW

**Comment by skyl** [comment]
utACK❗️

**Comment by nlopezgi** [comment]
Hi @achew22 ,
any chance you can finish this PR?
Anyone else interested in continuing this work?

File: k8s/object.bzl [hunk 1]
```diff
@@ -97,11 +97,21 @@ def _impl(ctx):
         image_chroot_arg = "$(cat %s)" % _runfiles(ctx, image_chroot_file)
         all_inputs += [image_chroot_file]
 
+    expanded_template = ctx.actions.declare_file(ctx.label.name + ".expanded.yaml")
```
[the remark this answers is not in the document]
Comment (achew22) [reply k8s/object.bzl]: Done

**Comment by achew22** [comment]
I've also added a test to demonstrate the new functionality. PTAL

**Comment by achew22** [comment]
*  [X] Make the template an explicit output of this rule
*  [X] create a sample yaml file with some template values
*  [X] Create a new k8s_object target in one of the /example/.. BUILD files, pass in your yaml file and the new substitutions attr
*  [X] add a file_test that checks the output of the k8s_object rule does include the modified template values.
*  [ ] add the file_test to our bazelci yaml (https://github.com/bazelbuild/rules_docker/blob/master/.bazelci/presubmit.yml)
    * Is this necessary? It looks like you test `...` from the root of the tree. I believe that would cover this, no?

**Comment by achew22** [comment]
Friendly ping. @nlopezgi could I have you take a look at this? It looks like the errors

```
ERROR: /workdir/examples/hellogrpc/py/server/BUILD:23:1: no such package '@py_image_base//image': Pull command failed: F0313 03:36:19.846622    1172 __main__.py:133] Error pulling and saving image gcr.io/distroless/python2.7@sha256:6d1384514d089f85d8649e5e332ef07f64e2e4aae43602d0f78ebbe42362f474: response: {'status': '503', 'content-length': '19', 'server': 'UploadServer', 'x-guploader-uploadid': 'AEnB2UriGOhtsuTZG6Pp8h2lysDCKgfVXhYy0pJwbp9KqEWjAuHM19subzyh0VaIDSnh2bY1sqVu7PNq7bzkzUWqWgR5KxFLhRO3hEYuGYbsQW6b4BIgxMQ', 'date': 'Wed, 13 Mar 2019 03:36:19 GMT', 'content-type': 'text/html; charset=UTF-8'}
``` 

are more from an old version of Ubuntu than this PR but I'm happy to revisit that claim if you think otherwise

**Comment by nlopezgi** [comment]
sorry for the delay @achew22 , thanks a lot for adding the test + doc!

**Pull request merged**